advancedlog/logger.go

```diff
@@ -0,0 +1,54 @@
+package logger
+
+import "os"
+
+// OutputFormat ...
+type OutputFormat string
+
+const (
+	// JSONFormat ...
+	JSONFormat OutputFormat = "json"
+)
+
+// Logger ...
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Info(args ...interface{})
+	Infof(format string, args ...interface{})
+	Done(args ...interface{})
+	Donef(format string, args ...interface{})
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Print(args ...interface{})
+	Printf(format string, args ...interface{})
+	Println(args ...interface{})
+	EnableDebugLog(enable bool)
+	IsDebugLogEnabled() bool
+}
+
+// SimplifiedLogger ...
+type SimplifiedLogger interface {
+	EnableDebugLog(enabled bool)
+	IsDebugLogEnabled() bool
+	LogMessage(producer Producer, level Level, message string)
+}
+
+// DefaultLogger ...
+var DefaultLogger = NewMainLogger()
+
+// SetOutputFormat ...
+func SetOutputFormat(outputFormat string) {
```

Comment:
`outputFormat` type could be `outputFormat` to ensure clients are using valid values.

Reply:
Updated in https://github.com/bitrise-io/go-utils/pull/166/commits/38b681bef8d4ec5b938ae7156c6515d6f3bb6cbb